spec.html

```diff
@@ -34306,17 +34306,17 @@ <h1>String.prototype.slice ( _start_, _end_ )</h1>
       </emu-clause>
 
       <emu-clause id="sec-string.prototype.split">
-        <h1>String.prototype.split ( _separator_, _limit_ )</h1>
-        <p>This method returns an Array into which substrings of the result of converting this object to a String have been stored. The substrings are determined by searching from left to right for occurrences of _separator_; these occurrences are not part of any String in the returned array, but serve to divide up the String value. The value of _separator_ may be a String of any length or it may be an object, such as a RegExp, that has a @@split method.</p>
+        <h1>String.prototype.split ( _separator_, _substringLimit_ )</h1>
```

Comment:
why is "limit" any less clear than "substringLimit" in a "string split" method?

Reply:
Some languages and libraries limit the count of _splits_ while others limit the count of _substrings_. This rename provides a hint that ECMAScript is in the latter category, as opposed to e.g. [Python](https://docs.python.org/3/library/stdtypes.html#str.split) (`len("foo bar baz".split(" ", 1)) == 2`) and [JMESPath](https://jmespath.site/main/#func-split) (``` length(split('foo bar baz', ' ', `1`)) == `2` ```) and [Swift](https://developer.apple.com/documentation/swift/int/words-swift.struct/split(separator:maxsplits:omittingemptysubsequences:)) (`"foo bar baz".split(separator: " ", maxSplits: 1).count == 2`).

Reply:
I understand the difference; I'm skeptical that a) the parameter name in the specification is the source of that confusion, and that fixing it will have any impact, and b) that the majority of newcomers to JS will necessarily be familiar with languages in the former category.

Reply:
I'm not claiming that renaming a parameter in the specification will fix whatever confusion exists in the ecosystem at large, but doing so will at least prevent creating unnecessary additional confusion when reading the spec.